### ISSUE TITLE
Access to Param.id

### ISSUE THREAD

**Issue by SeverinAlexB** [tracel-ai/burn#961]
**Describe the bug**

Hey guys! I am working on a very custom neural network for a web of trust use case (web of trust = feed forward nn). I train the nn on single samples and I need to apply the gradient with layer specific learning rates.
For that, I defined a feed forward network with a variable amount of layers.  Unfortunately, I struggle to extract the gradient.

```
type MyBackend = ADBackendDecorator<NdArrayBackend<f32>>;

pub struct FeedForward {
    linears: Vec<Linear<MyBackend>>
}
```

To train the network, I started to write this method.

```
impl FeedForward {
    pub fn train(&self, target_index: i64) -> () {
        let input: Tensor<MyBackend, 2> = Tensor::ones(Shape::new([1, 1])); // Always ones
        let mut x: Tensor<MyBackend, 2> = input;
        for i in 0..(self.linears.len() -1) {
            x = self.linears[i].forward(x);
            x = relu(x);
        }

        let x = self.linears[self.linears.len() - 1].forward(x);
        // Don't use softmax here because it screws up CrossEntropy.

        let targets: Tensor<MyBackend, 1, Int> = Tensor::from_data(Data::new(vec![target_index], Shape::new([1])));
        let loss: Tensor<MyBackend, 1> = CrossEntropyLoss::new(None).forward(x.clone(), targets.clone());
        println!("Loss: {}", loss);

        let gradient = loss.backward();

        /** Concentrate on 1 layer because
        * `GradientsParams::from_grads::<MyBackend, _>(gradient, self);` does not work 
        * due with `Vec<Linear<MyBackend>>`.
        */
        let layer = &self.linears[self.linears.len() - 1];
        let grads = GradientsParams::from_grads::<MyBackend, _>(gradient, layer);

        // No access to the ParamId to get the gradient.
        let grad_tensor = grads.get(&layer.weight.id);
    }
}
``` 

Turns out all properties of `Param` are visibility restricted.

```
pub struct Param<T> {
    pub(crate) id: ParamId,
    pub(crate) value: T,
}
```

Is there another way to extract the gradient or do I use the library incorrectly? I am new to burn so let me know if I make some obvious mistakes here :)

Btw: Great work you are doing here. I love the library so far!

**To Reproduce**
Run the provided code. Let me know if you need a anything more but I guess the issue is clear?

**Expected behavior**
Param.id should be `pub` instead of `pub(create)`.

**Comment by nathanielsimard** [comment]
To access the gradient of a tensor directly, you can use the gradients variable directly instead of using the `GradientsParams`.

```rust
let grad_w = self.weight.grad_remove(&mut gradients); // Pop the gradient
```

The `GradientsParams` has the benefit of being serializable, and its ID is consistent across runs, necessary for checkpoints or gradient accumulation, but not necessarily in all cases. Though I think we could still make the id public.

**Comment by SeverinAlexB** [comment]
Ahh, cool that is very helpful. Works like a charm. No need to make the ID public then from my side. Thx for your help.

**Comment by laggui** [comment]
Completed in PR #1859
